From: David Disseldorp <ddiss@suse.de>
Date: Wed, 15 May 2019 16:56:39 +0200
Subject: ceph: fix "ceph.dir.rctime" vxattr value
Git-commit: 718807289d4130be1fe13f24f018733116958070
Patch-mainline: v5.3-rc1
References: bsc#1148133 bsc#1135219

The vxattr value incorrectly places a "09" prefix to the nanoseconds
field, instead of providing it as a zero-pad width specifier after '%'.

Fixes: 3489b42a72a4 ("ceph: fix three bugs, two in ceph_vxattrcb_file_layout()")
Link: https://tracker.ceph.com/issues/39943
Signed-off-by: David Disseldorp <ddiss@suse.de>
Reviewed-by: Ilya Dryomov <idryomov@gmail.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/xattr.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/xattr.c
+++ b/fs/ceph/xattr.c
@@ -223,7 +223,7 @@ static size_t ceph_vxattrcb_dir_rbytes(s
 static size_t ceph_vxattrcb_dir_rctime(struct ceph_inode_info *ci, char *val,
 				       size_t size)
 {
-	return snprintf(val, size, "%ld.09%ld", (long)ci->i_rctime.tv_sec,
+	return snprintf(val, size, "%ld.%09ld", (long)ci->i_rctime.tv_sec,
 			(long)ci->i_rctime.tv_nsec);
 }
 
